src/history/file_backed.rs

```diff
@@ -137,6 +137,11 @@ impl History for FileBackedHistory {
             // The unwritten entries
             let own_entries = self.entries.range(self.len_on_disk..);
 
+            let parent = fname.parent().expect("history path shouldn't be '/'");
+            if !parent.exists() {
+                std::fs::create_dir_all(parent)?;
+            }
```

Comment:
I think to remember that `std::fs:created_dir_all()` is already defensive enough (like `mkdir -p`) to allow the folder to exist.
Basically you can take the code from `FileBackedHistory::with_file()` (Filesystem root would be valid as it always exists)
```suggestion
            if let Some(base_dir) = fname.parent() {
                std::fs::create_dir_all(base_dir)?;
            }
```